src/azure.js

```diff
@@ -52,17 +52,38 @@ function saveTenantIdToConfig(tenantId) {
 }
 
 function auth(tenantId) {
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout
+  });
+
   return new Promise((resolve, reject) => {
     msRestAzure.interactiveLogin({ domain: tenantId }, (err, credentials, subscriptions) => {
       if (err) {
         reject(err);
         return;
       }
-      resolve({ credentials, subscriptions });
+
+      // Assumption, we'll only ever need the selected subscription
+      displayChooseAzureSubscriptionMessage(subscriptions);
+
+      rl.question('Type the number of the subscription: ', subscriptionId => {
+        let id = subscriptionId ? (parseInt(subscriptionId)-1) : 0;
+        let subscription = subscriptions[id];
+        resolve({ credentials, subscription });
+      })
```

Comment:
It doesn't look like you're calling [readline.close()](https://nodejs.org/api/readline.html#readline_readline) on the interface.